### PR TITLE
Implement course list page

### PR DESCRIPTION
**Pull request by DeanCivlin**
**Added New Course List page** 
1. Created new course-list-page html&css files
   and implemented the desired UI flow and design
2. Added Relevant code to urls.py
3. Added new courseList func to views
4. Created new course-page html&css files.
   this was done to connect the relevant connections
   from course-list-page forms
   and as a setup for the implementation of the
   course page
5. Added relevant code to urls.py
6. Added course-list test to test_client.py

### CODE REVIEW

File: AcadeMe/views.py [hunk 1]
```diff
@@ -33,3 +33,19 @@ def add_navbar_links_to_context(request, context):
...
+    context['seluniversity'] = seluniversity
+    context['seldegree'] = seldegree
+    all_courses = Course.objects.all()
+    context['all_courses'] = all_courses
+    add_navbar_links_to_context(request, context)
+    return render(request, 'landing/course-list-page.html', context)
+
+
+def course(request):
+    return render(request, 'landing/course-page.html')
```
Comment (razashtar93) [inline AcadeMe/views.py:51]: The course-page.html present the all navbar links correctly?
if not please add a call to the function:  add_navbar_links_to_context(request, context)

**Review by razashtar93 — APPROVED** [review]
LGTM

File: AcadeMeData/tests/test_client.py [hunk 2]
```diff
@@ -11,3 +11,9 @@ def test_homepage(client):
 def test_contactus(client):
     response = client.get('/contact_us/')
     assert response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_course_list(client):
+    response = client.get('/course-list/')
+    assert response.status_code == 200
```
Comment (Yarboa) [inline AcadeMeData/tests/test_client.py:19]: You could test the context here
that is needed by the template
  context['seluniversity'] = 
    context['seldegree'] =

File: AcadeMe/views.py (hunk 1 above)
Comment (DeanCivlin) [reply AcadeMe/views.py:51]: Yap... I have implemented it on #128

File: AcadeMeData/tests/test_client.py (hunk 2 above)
Comment (DeanCivlin) [reply AcadeMeData/tests/test_client.py:19]: will do!

File: AcadeMeData/tests/test_views.py [hunk 3]
```diff
@@ -20,3 +20,16 @@ def test_contactus_html(client, user_example):
     assertTemplateUsed(response, 'landing/contact_us.html')
     assert response.context['navbar_links'] == {f"Welcome {user_example.user.username}": "#",
                                                 "Logout": "/logout"}
+
+
```
Comment (DeanCivlin) [inline AcadeMeData/tests/test_views.py:24]: @Yarboa  Here is the test I added to test the course list page context

File: AcadeMeData/tests/test_client.py (hunk 2 above)
Comment (DeanCivlin) [reply AcadeMeData/tests/test_client.py:19]: @Yarboa 
Done. The asked test can be found on `AcadeMeData/tests/test_views.py`

**Review by Yarboa — COMMENTED** [review]
LGTM, although it seems the idea if what to test and where is not fully understood
Please rebase

**Comment by DeanCivlin** [comment]
@Yarboa 
Rebase is done.

**Review by DanielItkes — APPROVED** [review]
Good work Dean!

**Comment by Yarboa** [comment]
Please rebase

**Review by shakedhagag — APPROVED** [review]
Impressive work Dean

**Comment by DeanCivlin** [comment]
@Yarboa
Rebase is done

**Review by Yarboa — APPROVED** [review]
LGTM,

**Pull request merged**